src/Zforce.cpp

```diff
@@ -1013,4 +1022,17 @@ void Zforce::ClearBuffer(uint8_t* buffer)
   memset(buffer, 0, MAX_PAYLOAD);
 }
 
+void Zforce::SerializeInt(uint16_t value, uint8_t* serialized)
```

Comment:
I would request that, for simplicity, don't return a void, return the length of the array. That way you don't have to count it manually elsewhere, and in the future, we can support larger values than 32767 by just changing the serializer method.

Reply:
And for future proofing, set it to an int32_t, as ints can in fact be positive and negative in asn.1.

Reply:
The code calling this would then possibly be just:

uint8_t serializedTime[2];
uint8_t timeLength = SerializeInt(time, serializedTime);

And the rest would be the same. serializedTime is on the stack, so it doesn't steal any heap memory, and is returned when the method ends. The compiler will most likely, due to the stack alignment, already align it on 16 or 32-bit boundary, so it just makes it easier.

Reply:
Absolutely. I have done the changes regarding the SerializeInt method and also pushed changes for the Click-On-Touch request message.